packages/xo-server/src/api/vm.mjs

```diff
@@ -237,6 +239,11 @@ export const create = defer(async function ($defer, params) {
     await this.allocIpAddresses(vif.$id, concat(vif.ipv4_allowed, vif.ipv6_allowed)).catch(() => xapi.deleteVif(vif))
   }
 
+  if (params.createVtpm) {
+    const vtpmRef = await xapi.VTPM_create({ VM: xapiVm.$ref })
+    $defer.onFailure(() => destroyVtpm.call(this, { vtpm: xapi.getObjectByRef(vtpmRef) }))
```

Comment:
Do not call API methods, they are intended for external use.

Either call the underlying method directly or move the logic somewhere else.